### PR TITLE
275817 - i18n - Use i18n for Modele resource

### PR DESCRIPTION
**Pull request by MBausson**

### CODE REVIEW

File: app/views/modeles/_enclosure_fields.html.erb [hunk 1]
```diff
@@ -1,30 +1,30 @@
...
         <%= f.hidden_field :_destroy %>
-        <%= link_to "Supprimer", '#', class: "remove_fields" %>
+        <%= link_to t("action.delete"), '#', class: "remove_fields" %>
       </span>
     </div>
     <ul id="enclosureList_<%= f.object.id %>" class="list-group panel-body">
       <div class="list-group-item filtered">
         <fieldset>
-          <%= f.label "Type d'affchage : " %>
+          <%= f.label t(".display") %>
```
Comment (nicolas-brousse) [inline app/views/modeles/_enclosure_fields.html.erb:13]:
```suggestion
          <%= f.label :display %>
```

To be manage in with ActiveRecord i18n translations

File: app/views/modeles/edit.html.erb [hunk 2]
```diff
@@ -1,21 +1,23 @@
 <% provide(:title, @modele.name) %>
 <%= render 'layouts/breadcrumb', breadcrumb_variables: {
-  'Paramètres' => modeles_url,
-  'Modèles' => modeles_path,
-  "#{@modele.try(:name).present? ? @modele.name : 'Nouveau modèle' }" => ''}
+  t("breadcrumb.settings") => modeles_url,
```
Comment (nicolas-brousse) [inline app/views/modeles/edit.html.erb:3]:
```suggestion
  t("inventories") => modeles_url,
```

File: app/views/modeles/index.html.erb [hunk 3]
```diff
@@ -1,6 +1,9 @@
-<% provide(:title, "Modèles") %>
+<% provide(:title, Modele.model_name.human.pluralize) %>
 
-<%= render 'layouts/breadcrumb', breadcrumb_variables: {'Paramètres' => modeles_url, 'Modèles' => ''} %>
+<%= render 'layouts/breadcrumb', breadcrumb_variables: {
+  t("breadcrumb.settings") => modeles_url,
```
Comment (nicolas-brousse) [inline app/views/modeles/index.html.erb:4]:
```suggestion
  t("inventories") => modeles_url,
```

File: app/views/modeles/index.html.erb [hunk 4]
```diff
@@ -27,9 +32,15 @@
...
               <tr>
                 <th></th>
-                <th>Modèle</th>
-                <th>Disposition</th>
-                <th>Baies</th>
+                <th>
+                  <%= Modele.model_name.human %>
+                </th>
+                <th>
+                  <%= t(".display") %>
```
Comment (nicolas-brousse) [inline app/views/modeles/index.html.erb:39]:
```suggestion
                  <%= Enclosure.human_attribute_name(:display) %>
```

File: app/views/modeles/new.html.erb [hunk 5]
```diff
@@ -1,16 +1,18 @@
 <%= render 'layouts/breadcrumb', breadcrumb_variables: {
-  'Paramètres' => modeles_url,
-  'Modèles' => modeles_path,
-  "#{@modele.try(:name).present? ? @modele.name : 'Nouveau modèle' }" => ''
+  t("breadcrumb.settings") => modeles_url,
```
Comment (nicolas-brousse) [inline app/views/modeles/new.html.erb:2]:
```suggestion
  t("inventories") => modeles_url,
```

File: app/views/pages/_params_menu.html.erb [hunk 6]
```diff
@@ -8,7 +8,7 @@
     <a href="<%= rooms_path %>" class="list-group-item <%= controller.controller_name == "rooms" ? "active" : "" %>"><%= Room.model_name.human.pluralize %></a>
     <a href="<%= islets_path %>" class="list-group-item <%= controller.controller_name == "islets" ? "active" : "" %>"><%= Islet.model_name.human.pluralize %></a>
     <a href="<%= bays_path %>" class="list-group-item <%= controller.controller_name == "bays" ? "active" : "" %>"><%= t(".location.bays") %></a>
-    <a href="<%= frames_path %>" class="list-group-item <%= controller.controller_name == "frames" ? "active" : "" %>"><%= Frame.model_name.human.pluralize %></a>
+    <a href="<%= frames_path %>" class="list-group-item <%= controller.controller_name == "frames" ? "active" : "" %>"><%= Frame.model_name.human %></a>
```
Comment (nicolas-brousse) [inline app/views/pages/_params_menu.html.erb:11]: @MBausson for which issue this has been removed?

File: app/views/modeles/_form.html.erb [hunk 7]
```diff
@@ -12,41 +12,41 @@
   <% end %>
 
   <div class="form-group">
-    <%= f.label 'Nom', class: "col-sm-2 col-md-1 control-label" %>
+    <%= f.label :name, class: "col-sm-2 col-md-1 control-label" %>
     <div class="col-sm-10 col-md-11">
       <%= f.text_field :name, class: "form-control" %>
     </div>
-    <%= f.label :category_id, class: "col-sm-2 col-md-1 control-label" %>
+    <%= f.label :category, class: "col-sm-2 col-md-1 control-label" %>
```
Comment (nicolas-brousse) [inline app/views/modeles/_form.html.erb:19]:
```suggestion
    <%= f.label :category_id, class: "col-sm-2 col-md-1 control-label" %>
```

File: app/views/modeles/_form.html.erb [hunk 8]
```diff
@@ -12,41 +12,41 @@
...
     <div class="col-sm-10 col-md-5">
       <%= f.collection_select(:category_id, Category.order(Arel.sql('LOWER(name)')), :id, :name, {}, {class: "form-control"}) %>
     </div>
-    <%= f.label :architecture_id, class: "col-sm-2 col-md-1 control-label" %>
+    <%= f.label :architecture, class: "col-sm-2 col-md-1 control-label" %>
```
Comment (nicolas-brousse) [inline app/views/modeles/_form.html.erb:23]:
```suggestion
    <%= f.label :architecture_id, class: "col-sm-2 col-md-1 control-label" %>
```

File: app/views/modeles/_form.html.erb [hunk 9]
```diff
@@ -12,41 +12,41 @@
...
     <div class="col-sm-10  col-md-5">
       <%= f.collection_select(:architecture_id, Architecture.order(Arel.sql('LOWER(name)')), :id, :name, {}, {class: "form-control"}) %>
     </div>
     <%= f.label :u, class: "col-sm-2 col-md-1 control-label" %>
     <div class="col-sm-10 col-md-5">
       <%= f.number_field :u, class: "form-control" %>
     </div>
-    <%= f.label :manufacturer_id, class: "col-sm-2 col-md-1 control-label" %>
+    <%= f.label :manufacturer, class: "col-sm-2 col-md-1 control-label" %>
```
Comment (nicolas-brousse) [inline app/views/modeles/_form.html.erb:31]:
```suggestion
    <%= f.label :manufacturer_id, class: "col-sm-2 col-md-1 control-label" %>
```

File: app/views/modeles/_form.html.erb [hunk 10]
```diff
@@ -12,41 +12,41 @@
...
     </div>
-    <%= f.label 'Couleur', class: "col-sm-2 col-md-1 control-label" %>
+    <%= f.label :color, class: "col-sm-2 col-md-1 control-label" %>
     <div class="col-sm-10 col-md-5">
       <%= f.text_field :color, class: "form-control", style: "background-color:#{f.object.color};" %>
     </div>
   </div>
   <div>
-    <%= content_tag :span, 'Composants', class: "col-sm-2 control-label" %>
+    <%= content_tag :span, Modele.human_attribute_name(:enclosures), class: "col-sm-2 control-label" %>
```
Comment (nicolas-brousse) [inline app/views/modeles/_form.html.erb:45]:
```suggestion
    <%= content_tag :span, Enclosure.model_name.human.pluralize, class: "col-sm-2 control-label" %>
```

File: app/views/modeles/_enclosure_fields.html.erb [hunk 11]
```diff
@@ -1,30 +1,30 @@
...
           <%= f.select :display, options_for_select([:vertical, :horizontal, :grid], f.object.display) %>
         </fieldset>
       </div>
       <div class="list-group-item filtered">
         <fieldset>
-          <%= f.label "Grid Areas :" %>
+          <%= f.label t(".grid_areas") %>
```
Comment (nicolas-brousse) [inline app/views/modeles/_enclosure_fields.html.erb:19]:
```suggestion
          <%= f.label :grid_areas %>
```

To be manage in with ActiveRecord i18n translations

**Review by nicolas-brousse — CHANGES_REQUESTED** [review]
@MBausson you forgot to move string for controller flash messages

File: app/views/modeles/_composant_fields.html.erb [hunk 12]
```diff
@@ -33,7 +33,7 @@
           <span class="glyphicon glyphicon-resize-vertical"></span>
         </span>
       <% end %>
-      <%= f.text_field :name, { class: 'form-control', placeholder: 'Nom' } %>
+      <%= f.text_field :name, { class: 'form-control', placeholder: t(".name") } %>
```
Comment (nicolas-brousse) [inline app/views/modeles/_composant_fields.html.erb:36]: @MBausson Use existing rails way

in `activerecord.fr.yml`

```
  helpers:
    placeholder:
      modele:
        name: 'Nom'
```

And in this file
```suggestion
      <%= f.text_field :name, { class: 'form-control', placeholder: true } %>
```

File: config/locales/activerecord.fr.yml [hunk 13]
```diff
@@ -127,6 +128,20 @@ fr:
...
+        name: Nom
+        category: Catégorie
+        category_id: Catégorie
+        architecture: Architecture
+        architecture_id: Architecture
+        manufacturer: Fabricant
+        manufacturer_id: Fabricant
+        nb_elts: Nb d'éléments
+        color: Couleur
+        enclosures: Composants
```
Comment (nicolas-brousse) [inline config/locales/activerecord.fr.yml:141]: @MBausson is this still used?

File: config/locales/fr.yml [hunk 14]
```diff
@@ -282,6 +282,24 @@ fr:
...
+  modeles:
+    form:
+      add_enclosure: Ajouter une enclosure
+    enclosure_fields:
+      enclosure_position: Enclosure %{position}
+      new_enclosure: Nouvelle enclosure
+      grid_areas: "Grid areas :"
+      add_slot: Ajouter un slot
+    composant_fields:
+      name: Nom
```
Comment (nicolas-brousse) [inline config/locales/fr.yml:294]: @MBausson is this still used?

File: config/locales/fr.yml [hunk 15]
```diff
@@ -282,6 +282,24 @@ fr:
...
+    index:
+      add_model: Ajouter un modèle
+      delete_confirmation: Ce modèle ne pourra pas être supprimé tant qu'il sera associé à au moins un serveur ou une enclosure. Voulez-vous continuer ?'
+      display: Disposition
```
Comment (nicolas-brousse) [inline config/locales/fr.yml:298]: @MBausson is this still used?

**Review by nicolas-brousse — CHANGES_REQUESTED** [review]

File: app/views/modeles/_composant_fields.html.erb (hunk 12 above)
Comment (MBausson) [reply app/views/modeles/_composant_fields.html.erb:36]: I can't achieve to translate the placeholder using this way, the placeholder stays in english ..

File: config/locales/activerecord.fr.yml (hunk 13 above)
Comment (MBausson) [reply config/locales/activerecord.fr.yml:143]: Yes

File: config/locales/fr.yml (hunk 15 above)
Comment (MBausson) [reply config/locales/fr.yml:319]: display is no longer used

File: config/locales/activerecord.fr.yml (hunk 13 above)
Comment (nicolas-brousse) [reply config/locales/activerecord.fr.yml]: Why not using `Enclosure.model_name.human`?

**Pull request merged**